src/lib/viewers/BaseViewer.js

```diff
@@ -1038,17 +1040,23 @@ class BaseViewer extends EventEmitter {
         this.annotator.scrollToAnnotation(data);
     }
 
-    handleAnnotationsOnLoad = () => {
+    /**
+     * Handles retrieving the active annotation id from a deep-linked annotation and scrolls to the annotation.
+     * @param {Object[]} data - annotations array from emitted from box-annotations.
+     * @return {void}
+     */
+    handleAnnotationsOnLoad = (data = []) => {
         const {
             file: { id },
         } = this.options;
 
         const loadedAnnotationId = getProp(this.options, `fileOptions.${id}.annotations.activeId`, null);
-        const annotation = loadedAnnotationId ? this.annotator.getAnnotationById(loadedAnnotationId) : null;
 
-        if (annotation) {
-            this.handleScrollToAnnotation(annotation);
+        if (!loadedAnnotationId) {
+            return;
         }
+
+        this.handleScrollToAnnotation(data.find(entry => entry.id === loadedAnnotationId));
```

Comment:
Should we make sure an annotation is found before calling into `this.handleScrollToAnnotation`? Otherwise it looks like we call into the `this.annotator.scrollToAnnotation` with `undefined`